### ISSUE TITLE
Is there a way to use on Desktop?

### ISSUE THREAD

**Issue by asfaltboy** [Ayagikei/LifeUp#12]
If there is no web/desktop client, I imagine we could perhaps allow sync out to a [todo.txt](http://todotxt.org/) list perhaps?

**Comment by Ayagikei** [comment]
Hello asfaltboy,
Thanks for the question.

There is currently no version other than the Android platform.
We just created the flutter version repository.
However, we still need a lot of time to build it up.

Even if we develop multiple platforms, due to data storage models, technical limitations, and server resources, it is still difficult for us to achieve incremental synchronization functions like commercial works.

Thanks for the suggestion, we will consider adding the function of exporting a simple To-Do.txt file.

The application currently supports exporting complete data (pictures, databases).
We also plan to develop a function in the future to support users to choose what they want to export into a .json file or directly share it to the market module.
As for incremental synchronization, we may have to wait for the app to develop and grow enough to be able to consider...

**Comment by cr7kaka10** [comment]
作者可以参考一下logseq，可以通过百度网盘工作空间，或者坚果云，实时同步本地文件，然后另一台设备同步下来数据后，就可以显示最新的数据了

**Comment by Ayagikei** [comment]
> 作者可以参考一下logseq，可以通过百度网盘工作空间，或者坚果云，实时同步本地文件，然后另一台设备同步下来数据后，就可以显示最新的数据了

感谢提供参考。还没怎么看代码，不太熟悉（
我们这边也发现了一堆支持使用Google Drive/WebDAV做增量同步的开源软件或者商业软件，比如Journey这款日记app就是使用Google Drive实现的增量同步。还有一款开源笔记APP（https://github.com/KumarManas04/NotesSync）。

简单看了一下，有一些收获是：他们是把每一篇日记/笔记作为一个文件进行存储，并且文件里记录下创建和编辑时间。图片附件也是作为独立的文件存储。
这样确实可以使用Google Drive/WebDAV等支持文件遍历检索的渠道实现增量同步。

但他们有个特色是，他们的数据库关系比较简单。（基本都是一篇日记对应多张图片，然后可能就没复杂结构了）。
现在《人升》的数据模型其实错综复杂，比如“任务”可能会关联上“商品”、“成就”、“图片附件”、“感想”甚至是“金币”和“经验值”的获得记录。在这种情况下，我还没想出来怎么将这种关联关系抽象成一份份文件进行同步，比较头疼。

---
**In English:**

> The author can refer to logseq. You can synchronize local files in real-time through Baidu Netdisk Workspace or Nuts Cloud, and then another device can synchronize the data, and the latest data can be displayed.

Thanks for the reference. Haven't looked at the code much, don't know much about it (
This time we also found open source software or commercial software that successfully supports the use of Google Drive/WebDAV and incremental synchronization, such as the incremental synchronization of the Journey application using Google Drive And an open-source app: (http//github.com/KumarManas04/NotesSync).

Take a brief look, there are some gains: They store each diary/note as a file, and the creation and editing time are recorded in the file. Picture attachments are also stored as separate files.
In this way, it is indeed possible to use Google Drive/WebDAV file traversal and retrieval capabilities to achieve such incremental support synchronization.

But their database relationships are simple. (Basic picture, then a possible set of complex structures).

Now the data model of *LifeUp* is actually very complicated.
For example, "task" may be associated with "shop items", "achievements", "picture attachments", "feelings" and even "coin" and "experience value" acquisition records .
Under these circumstances, I haven't figured out how to abstract this relationship into a file for synchronization, which is troublesome.

**Comment by kotstoiki** [comment]
Another thing that's possible for now, considering you don't have Windows 11 (like me) is downloading a mobile simulator like BlueStacks. All you have to do after that is backing up the information on your phone e.g. on Dropbox  and restore it on the "PC's" app.

**Comment by FraYoshi** [comment]
I would personally prefer a web application before everything.
I run Linux and as much as I have all the possibilities to run in a VM or wine.. I would rather not, as it takes precious resources and discharges rapidly the battery when on mobile.
A far better solution, which would allow any OS to use it, is in my opinion, a web app or a local server on the smartphone to which users can connect to by inputting the address and port.
Money Manager Expense & Budget by Realbyte Inc. does exactly this, take a sneak peak for some inspiration 🙂

Link to the app:
https://play.google.com/store/apps/details?id=com.realbyteapps.moneymanagerfree

**Comment by Ayagikei** [comment]
> I would personally prefer a web application before everything. I run Linux and as much as I have all the possibilities to run in a VM or wine.. I would rather not, as it takes precious resources and discharges rapidly the battery when on mobile. A far better solution, which would allow any OS to use it, is in my opinion, a web app or a local server on the smartphone to which users can connect to by inputting the address and port. Money Manager Expense & Budget by Realbyte Inc. does exactly this, take a sneak peak for some inspiration 🙂
> 
> Link to the app: https://play.google.com/store/apps/details?id=com.realbyteapps.moneymanagerfree

Hi, thanks for your thoughts and feedback! In fact, this is what our bar LAN desktop does.

You can check [https://github.com/Ayagikei/LifeUp-Desktop](https://github.com/Ayagikei/LifeUp-Desktop).

And it is based on compose-multiplatform, and theoretically also supports Windows, Mac OS, and Linux.

> We have not released a Linux installation package yet, but it should be able to run directly from source code compilation.

Compose-web may have more development and technical limitations on this basis, and the above system has covered most scenarios, so we may not give priority to web yet.

**Comment by FraYoshi** [comment]
> You can check https://github.com/Ayagikei/LifeUp-Desktop

Yes, it seems a great project, but have zero knowledge of IntelliJ.. don't even know where to start.. :' )
I build the project with the IntelliJ IDE, but then I'm stuck _._

I guess I just need to study a bit of this programming language before I can continue... A little build guide would be greatly appreciated :)

Anyway, keep it up the great work!! ✨

**Comment by Ayagikei** [comment]
> > You can check https://github.com/Ayagikei/LifeUp-Desktop
> 
> Yes, it seems a great project, but have zero knowledge of IntelliJ.. don't even know where to start.. :' ) I build the project with the IntelliJ IDE, but then I'm stuck _._
> 
> I guess I just need to study a bit of this programming language before I can continue... A little build guide would be greatly appreciated :)
> 
> Anyway, keep it up the great work!! ✨

Thanks for trying!

If your development environment works, you can try to run the project directly and execute the main method. If there are no exceptions or compatibility issues, you should be able to see the desktop UI.

![image](https://github.com/Ayagikei/LifeUp/assets/29143057/8cbf4414-d1d6-4ad7-9779-1a60df735d11)

If you want to package, you can also execute the command through the gradle panel:

![image](https://github.com/Ayagikei/LifeUp/assets/29143057/e9f3ac84-9850-4d88-800f-644b54f53746)



But it doesn't matter if you don't do this, we will try to build a Linux environment later. After our testing and verification, we will also try to release the Linux installation package.

**Thank you for your support and understanding!** 💕

**Comment by FraYoshi** [comment]
Writing to report that running the way you explained, worked well!
I have Tasks, Status, Shop (not inventory), Achievements, and Feelings EXPORTABLE ❤️

![2023-06-28_21-47-02](https://github.com/Ayagikei/LifeUp/assets/14005514/1926a89a-a597-4093-9cf8-8d4efd7a6905)

Although, have not been successful on building for my Linux distribution (Arch), but that's not a problem, I can run It the manual way, and this is more than enough for me! Having a desktop app, makes the experience stand out from the rest of the gamification apps, thanks very much! ❤️

![2023-06-28_21-58-15](https://github.com/Ayagikei/LifeUp/assets/14005514/0fac9d05-10d9-40d5-adb5-1b83580100a6)

**Comment by depocoder** [comment]
@Ayagikei I think it would be great if you create web version and extension for pomodoro like that https://habitica.fandom.com/wiki/Habitica_Pomodoro_SiteKeeper

**Comment by dmgolembiowski** [comment]
At some point within the next 12 months, I'll be integrating Super Productivity with LifeUp SDK's HTTP API. Super Productivity's mobile and desktop UI/UX is more mature and has the potential to beautifully hook into LifeUp's ecosystem. All that's missing is oneway bridging glue code for synchronizing each direction.
